release-20.2: vendor: bump Pebble to e9a2c44e0bf1

jbowens opened the pull request:
```
e9a2c44e db: close previous WAL before linking next WAL
```

Fixes #54142.

Release note (bug fix): Fixes a bug whereby a crash during WAL rotation
could cause Cockroach to error on restart reporting corruption.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54185)
<!-- Reviewable:end -->

itsbilal reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54185#-:-MGsdukZDklzpRpbuegZ:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @jbowens)

___
*[go.mod, line 154 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/54185#-MGsfR8G3eLvGnKXlkUg:-MGsfR8G3eLvGnKXlkUh:bvssylq) ([raw file](https://github.com/cockroachdb/cockroach/blob/27a201c0fa7cebfe487e0ba74482aa5b135dfe9b/go.mod#L154)):*
> ```XML
> 	golang.org/x/perf v0.0.0-20180704124530-6e6d33e29852
> 	golang.org/x/sync v0.0.0-20190911185100-cd5d95a43a6e
> 	golang.org/x/sys v0.0.0-20200909081042-eff7692f9009
> ```

I think you'll have to revert these changes again. I've started to manually revert all the hunks other than the Pebble one after running `go get -u`, or else the build doesn't succeed.


<!-- Sent from Reviewable.io -->

jbowens commented:
> I think you'll have to revert these changes again. I've started to manually revert all the hunks other than the Pebble one after running go get -u, or else the build doesn't succeed.

Yeah, I'll do that again for this PR. I filed #54193 for fixing the incompatibility with the current version of golang.org/x/sys.

jbowens committed:
vendor: bump Pebble to e9a2c44e0bf1

```
e9a2c44e db: close previous WAL before linking next WAL
```

Release note (bug fix): Fixes a bug whereby a crash during WAL rotation
could cause Cockroach to error on restart reporting corruption.
